### PR TITLE
Export rds

### PR DESCRIPTION
**Pull request by russHyde**
! Breaking changes !

The previous implementation of `treeview()/create_trees()/save_trees()` created a series of htmlwidgets and htmlwidget files (e.g., "tree-logistic_growth_rate-2022-11-09.html") that were to be imported to an iframe in the tfpbrowser shiny app. But, these htmlwidgets could readily be constructed inside the tfpbrowser app - and if they were, they could be incorporated without iframes.

Here, rather than saving the htmlwidget/html files, we save the ggtree objects from which those htmlwidgets can be created as ".rds" files (default pattern: "tree-logistic_growth_rate.rds" or "tree-clock_outlier.rds").

`create_trees` no longer makes the htmlwidgets and `save_trees` no longer saves the htmlwidgets.
A function `create_widget` is present in the source code - this was used for converting ggtree objects to htmlwidgets, but now it is only used in `plot_cluster_sina()` to convert scatter plots to htmlwidgets. That function could similarly be rewritten to only output ggplot objects (the corresponding htmlwidgets are not currently used in tfpbrowser AFAIK).

Note that the output from `save_trees()` now does not have the date in the filepath.

Due to the braking changes, and because we aren't yet at a 1.x release, the package version has been bumped to 0.2.0.

If requested, I can rewrite this so that the breaking changes are introduced in a trivial PR separately from the refactorings / tests / new (non-breaking) features that are introduced here.

### CODE REVIEW

File: R/treeview.R [hunk 1]
```diff
@@ -265,14 +265,7 @@ treeview <- function(e0,
...
   suppressWarnings({
-    plot_cluster_sina(
-      pldf,
-      output_dir = output_dir,
-      mut_regexp = mutations,
-      lineage_regexp = lineages
-    )
-
-    for (vn in setdiff(branch_cols, c("logistic_growth_rate"))) {
+    for (vn in unique(c("logistic_growth_rate", branch_cols))) {
```
Comment (jr-nicola) [inline R/treeview.R:268]: where does the change `setdiff()` to `unique()` come from? These don't seem to give the same values for me:

```
branch_cols <- c("A", "B", "C")
unique(c("A", branch_cols)) # returns "A" "B" "C"
setdiff(branch_cols, "A") # returns "B" "C"
```

Comment (russHyde) [reply R/treeview.R:268]: This reduces the following:
```
f("a")

for (x in setdiff(some_vec, "a")) {
  f(x)
}
```

To this:

```
for (x in unique("a", some_vec)) {
  f(x)
}
```

**Comment by jr-nicola** [comment]
Overall look good - left a small query :) A couple of other things:

I added in the .rds files saved from the new `treeview()` function and get the following warning in the console (although appears to render okay):
> ℹ Did you forget to specify a `group` aesthetic or to convert a numerical variable into a factor?
> Warning: The following aesthetics were dropped during statistical transformation: node, parent
I also get that warning when loading just the ggtree object, without {igraph} as well. `node` is a key parameter, at the moment this means that the cluster ID (node) that appears in the tooltip doesn't match what is selected when you click on the plot. I could come up with a (horrible) workaround, but ideally this is something that can be fixed in {tfpscanner} - happy for this to be a separate PR.

* Ideally we'd also make the mouseover/tooltip have only the cluster ID but I'm happy for that to be a separate pull request, to keep changes here mostly to a refactor.

* `plot_cluster_sina()` will likely need to be output as .rds files as part of WP5, when we think about how to include these in the app (format of plot is very different to other treeview plots). Again happy to leave this for a future PR.

File: R/treeview.R (hunk 1 above)
Comment (jr-nicola) [reply R/treeview.R:268]: Thanks - that makes sense

**Review by jr-nicola — APPROVED** [review]

**Comment by russHyde** [comment]
Ah, that might explain why there are the suppressWarnings calls around plot_tree(). I'll investigate later.

**Comment by russHyde** [comment]
If I remove the suppressWarnings({}) around the calls to create_trees() I do get some warnings, but I don't get that specific warning. The warnings observed are unaltered by inclusion/exclusion of the call to create the htmlwidget for the tree.

```
Warning messages:
1: The type of "continuous" argument was changed (v>=2.5.2). Now, it should be one of "color" (or "colour"), "size", "all",
and "none". 
2: It was set to TRUE, it should be replaced with "color" (or "colour"), this meaning the aesthethic of "color" (or "colour")
is continuous. 
3: The type of "continuous" argument was changed (v>=2.5.2). Now, it should be one of "color" (or "colour"), "size", "all",
and "none". 
4: It was set to TRUE, it should be replaced with "color" (or "colour"), this meaning the aesthethic of "color" (or "colour")
is continuous.
```

**Comment by russHyde** [comment]
I think this is due to a call to ggtree::geom_tree https://github.com/YuLab-SMU/ggtree/blob/6a7815201a62d89e8b9e76a38bbc4dfcd8ce52ec/R/geom_tree.R#L42

**Comment by jr-nicola** [comment]
That's interesting - I have a workaround to get the htmlwidget working as a shiny input even without the node column being assigned to data_id, and I can suppress the warnings in side the app as well. I'll have a look into this a bit more on the {tfpbrowser} side (I suspect it may be related to it being a ggtree object, and ggtree is currently not used in the app, so it might not understand what parents and nodes mean). Otherwise, happy for this to be merged whenever :)

**Comment by russHyde** [comment]
The original warning is thrown by ggplot2::Stat$compute_panel() https://github.com/tidyverse/ggplot2/blob/63125db184b6e7d5516007574b5d15acbfd8f470/R/stat-.r#L171

I'm not quite sure how to debug that....

**Pull request merged**